crates/storage/provider/src/traits/state.rs

```diff
@@ -226,3 +231,17 @@ pub trait BundleStateDataProvider: Send + Sync {
     /// Needed to create state provider.
     fn canonical_fork(&self) -> BlockNumHash;
 }
+
+/// A helper trait for [BundleStateWithReceipts] write state data to storage.
```

Comment:
nit: we should prob document that this is state + receipts (and later prob make these 2 separate things)

Reply:
agree, I wanted to keep diff small and keep function as is, but yeah, still should be separated